food-and-agriculture-organization-of-the-united-nations.csl

```diff
@@ -10,196 +10,368 @@
+      <date-part name="day"/>
+      <date-part name="month"/>
+      <date-part name="year"/>
+    </date>
   </locale>
+  <!-- Macros -->
   <macro name="editor">
-    <names variable="editor" delimiter="," suffix=".">
+    <!-- XX ed. / XX & XX eds. -->
+    <names variable="editor">
       <name and="symbol" delimiter-precedes-last="never" initialize-with="."/>
       <label form="short" strip-periods="false" prefix=", "/>
     </names>
   </macro>
   <macro name="author">
-    <names variable="author" suffix=". ">
-      <name and="symbol" delimiter-precedes-last="never" initialize-with="." name-as-sort-order="all"/>
-      <label form="short" prefix=" "/>
-      <substitute>
-        <names variable="editor"/>
-        <names variable="translator"/>
-        <text variable="title"/>
-      </substitute>
-    </names>
+    <choose>
+      <if match="any" variable="author editor translator">
+        <names variable="author">
+          <name and="symbol" delimiter-precedes-last="never" initialize-with="." name-as-sort-order="all"/>
+          <label form="short" prefix=", "/>
+          <substitute>
+            <names variable="editor"/>
+            <names variable="translator"/>
+          </substitute>
+        </names>
+      </if>
+      <else>
+        <text term="anonymous" text-case="capitalize-first"/>
+      </else>
+    </choose>
   </macro>
   <macro name="author-short">
-    <names variable="author">
-      <name form="short" and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never" et-al-min="4" et-al-use-first="1" initialize-with="."/>
-      <et-al font-style="italic"/>
-      <substitute>
-        <names variable="editor"/>
-        <names variable="translator"/>
-        <text variable="title"/>
-      </substitute>
-    </names>
+    <choose>
+      <if match="any" variable="author editor translator">
+        <names variable="author">
+          <name form="short" and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never" et-al-min="4" et-al-use-first="1" initialize-with="."/>
+          <et-al font-style="italic"/>
+          <substitute>
+            <names variable="editor"/>
+            <names variable="translator"/>
+          </substitute>
+        </names>
+      </if>
+      <else>
+        <text term="anonymous" text-case="capitalize-first"/>
+      </else>
+    </choose>
   </macro>
-  <macro name="publisher">
-    <group prefix=" ">
-      <text variable="publisher-place" suffix=","/>
-      <text variable="publisher" prefix=" " suffix="."/>
+  <macro name="date">
+    <!-- date format: year / undated / put "forthcoming" or else in "Date" field when needed.-->
+    <choose>
+      <if match="none" variable="issued">
+        <text term="no date"/>
+      </if>
+      <else>
+        <date date-parts="year" form="text" variable="issued"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-and-place">
+    <!-- place / place, publisher -->
+    <group>
```

Comment:
I put it this way because sometimes there is only publisher-place but no publisher. In this case, the output will be "publisher-place." (ending with a period), not "publisher-place," (ending with a comma, which is undesired).